pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/design/AnnotationsShouldBePascalCaseRule.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.rule.design;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import net.sourceforge.pmd.lang.apex.ast.ASTAnnotation;
+import net.sourceforge.pmd.lang.apex.ast.ASTField;
+import net.sourceforge.pmd.lang.apex.ast.ASTMethod;
+import net.sourceforge.pmd.lang.apex.ast.ASTModifierNode;
+import net.sourceforge.pmd.lang.apex.ast.ASTUserClass;
+import net.sourceforge.pmd.lang.apex.ast.ASTUserInterface;
+import net.sourceforge.pmd.lang.apex.ast.ApexNode;
+import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
+import net.sourceforge.pmd.lang.rule.RuleTargetSelector;
+
+/**
+ * Rule that checks if Apex annotations are written in PascalCase. This rule
+ * ensures that all annotations follow the standard naming convention where each
+ * word in the annotation name starts with a capital letter.
+ */
+public class AnnotationsShouldBePascalCaseRule extends AbstractApexRule {
+
+    private static final Set<Class<? extends ApexNode<?>>> VALID_PARENT_TYPES = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(ASTUserClass.class, ASTUserInterface.class, ASTMethod.class, ASTField.class)));
+
+    @Override
+    public Object visit(ASTAnnotation annotation, Object data) {
+        if (annotation.isResolved() && isValidAnnotationContext(annotation) && !isPascalCase(annotation)) {
+            asCtx(data).addViolation(annotation);
```

Comment:
We could even provide a specific message like: `The annotation @future should be in PascalCase: @Future`:

```suggestion
            asCtx(data).addViolation(annotation, annotation.getRawName(), annotation.getName());
```

And down below when defining the rule, we need to use this message:

```
message="The annotation @{0} should be in PascalCase: @{1}"
```

Then we can already see in the rule violation message the problem and the solution.